plugins/processors/parser/parser.go

```diff
@@ -54,38 +56,12 @@ func (p *Parser) Apply(metrics ...telegraf.Metric) []telegraf.Metric {
 
 		// parse fields
 		for _, key := range p.ParseFields {
-			for _, field := range metric.FieldList() {
-				if field.Key != key {
-					continue
-				}
-				value, err := p.toBytes(field.Value)
-				if err != nil {
-					p.Log.Errorf("could not convert field %s: %v; skipping", key, err)
-					continue
-				}
-				fromFieldMetric, err := p.parser.Parse(value)
-				if err != nil {
-					p.Log.Errorf("could not parse field %s: %v", key, err)
-					continue
-				}
-
-				for _, m := range fromFieldMetric {
-					// The parser get the parent plugin's name as
-					// default measurement name. Thus, in case the
-					// parsed metric does not provide a name itself,
-					// the parser  will return 'parser' as we are in
-					// processors.parser. In those cases we want to
-					// keep the original metric name.
-					if m.Name() == "" || m.Name() == "parser" {
-						m.SetName(metric.Name())
-					}
-				}
+			newMetrics = append(newMetrics, p.parseField(key, metric, false)...)
+		}
 
-				// multiple parsed fields shouldn't create multiple
-				// metrics so we'll merge tags/fields down into one
-				// prior to returning.
-				newMetrics = append(newMetrics, fromFieldMetric...)
-			}
+		// parse base64 fields
+		for _, key := range p.Base64Fields {
+			newMetrics = append(newMetrics, p.parseField(key, metric, true)...)
 		}
```

Comment:
I think you should do
```golang
		// parse fields
		for _, field := range metric.FieldList() {
			plain := slices.Contains(field.Key, p.ParseFields)
			b64 := slices.Contains(field.Key, p.Base64Fields)

			if !plain && !b64 {
				continue
			}
			if plain && b64 {
				p.Log.Errorf("field %s is listed in both plain and base64 fields; skipping", field.Key)
				continue
			}

			value, err := p.toBytes(field.Value)
			if err != nil {
				p.Log.Errorf("could not convert field %s: %v; skipping", key, err)
				continue
			}
	
			if b64 {
				decoded := make([]byte, base64.StdEncoding.DecodedLen(len(raw)))
				n, err := base64.StdEncoding.Decode(decoded, raw)
				if err != nil {
					p.Log.Errorf("could not decode base64 field %s: %v; skipping", field.Key, err)
					continue
				}
				value = decoded[:n]
			}

			fromFieldMetric, err := p.parser.Parse(value)
			if err != nil {
				p.Log.Errorf("could not parse field %s: %v", field.Key, err)
				continue
			}

			for _, m := range fromFieldMetric {
				// The parser get the parent plugin's name as
				// default measurement name. Thus, in case the
				// parsed metric does not provide a name itself,
				// the parser  will return 'parser' as we are in
				// processors.parser. In those cases we want to
				// keep the original metric name.
				if m.Name() == "" || m.Name() == "parser" {
					m.SetName(metric.Name())
				}
			}

			// multiple parsed fields shouldn't create multiple
			// metrics so we'll merge tags/fields down into one
			// prior to returning.
			newMetrics = append(newMetrics, fromFieldMetric...)
		}
```
and leave the other functions as is...

Reply:
Good suggestion, applied!